synapse/federation/federation_base.py

```diff
@@ -78,6 +78,7 @@ def _check_sigs_and_hashes(
 
         ctx = current_context()
 
+        @defer.inlineCallbacks
```

Comment:
I mentioned this in the commit, but it is sad to add an inlineCallbacks back in, but without this we need to convert a huge stack to async/await. I'd rather do that as a follow-up.